### PR TITLE
Patch 3

### PR DESCRIPTION
**Pull request by praxder**
This PR expands the docs for the `identifier` type of pattern matching to include extra, helpful info.

### CODE REVIEW

**Comment by praxder** [comment]
Note: this is in reference to issue: https://github.com/dart-lang/site-www/issues/4921

File: examples/language/lib/patterns/switch.dart [hunk 1]
```diff
@@ -26,10 +26,10 @@ void main() {
   }
   // #enddocregion constant-pattern
 
+  // #docregion list-pattern
   const a = 'a';
   const b = 'b';
   var obj = Object();
```
Comment (MaryaBelanger) [inline examples/language/lib/patterns/switch.dart:32]: Thanks for doing this! Could you move the `obj` declaration line outside of the `docregion`? That part is pretty arbitrary.

```suggestion
  var obj = Object();
  // #docregion list-pattern
  const a = 'a';
  const b = 'b';
```

Comment (praxder) [reply examples/language/lib/patterns/switch.dart]: Sounds good. Would you also like me to remove it from the 'src/language/patterns.md' file too?

Comment (parlough) [reply examples/language/lib/patterns/switch.dart]: Thanks! Yes please :)

The markdown snippet is matched against the `.dart` file to make sure they match.

Comment (praxder) [reply examples/language/lib/patterns/switch.dart]: done. :)

**Review by MaryaBelanger — APPROVED** [review]
Thank you! I'll take care of the build failures, but otherwise this is as good as landed

**Pull request merged**